spec/authorization_spec.rb

```diff
@@ -11,8 +11,14 @@
   let(:authorization) { client.authorize(domain: 'example.org')}
 
   context '#http01' do
-    it 'returns a HTTP01 object', vcr: { cassette_name: 'authorization_http_01' }  do
+    it 'returns a HTTP01 object', vcr: { cassette_name: 'authorization' }  do
       expect(authorization.http01).to be_a(Acme::Resources::Challenges::HTTP01)
     end
   end
+
+  context '#dns01' do
+    it 'returns a DNS01 object', vcr: { cassette_name: 'authorization' }  do
```

Comment:
Yes, both cassettes would contain the same stuff since the server responds with all challenges available.